### PR TITLE
make cheat codes clang-15 compatible

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

**Comment by kraflab** [comment]
I always wondered how these functions even work 😄

**Comment by fabiangreffrath** [comment]
Indeed! This is just another case of int/char* mixup that might have worked in the 1990s. I wonder how this could ever slip through in 64-bit builds. :thinking:

File: src/m_cheat.c [hunk 1]
```diff
@@ -403,13 +403,13 @@ static void cheat_notarget()
     plyr->message = "Notarget Mode OFF";
 }
 
-static void cheat_tst()
+static void cheat_tst(cheatarg_t arg)
 { // killough 10/98: same as iddqd except for message
-  cheat_god();
+  cheat_god((cheatarg_t){.i = 0});
```
Comment (rfomin) [inline src/m_cheat.c:408]: C89 doesn't support designated initializers, but I think we can move on to at least C99.

**Review by rfomin — APPROVED** [review]

File: src/m_cheat.c (hunk 1 above)
Comment (fabiangreffrath) [reply src/m_cheat.c:408]: How would this look in order to be C89 compliant?

Comment (fabiangreffrath) [reply src/m_cheat.c:408]: Like this?
https://github.com/fabiangreffrath/woof/blob/b12fa74d13228e54a20323eedfe5543d3a8a6550/src/m_misc.c#L115-L120

Comment (rfomin) [reply src/m_cheat.c:408]: Yes, but we still use `stdint.h` types, maybe I should remove this macro.

Comment (fabiangreffrath) [reply src/m_cheat.c:408]: So, by including `stdint.h` we have already left C89 terrain towards C99? Well, then I think it's time to let go. :wink:

**Pull request merged**